### PR TITLE
io.github.iSlammedMyKindle.godotGem

### PR DESCRIPTION
**Pull request by iSlammedMyKindle**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - `device=all` - required for game controller inputs. Also being used to enable the graphics card
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by iSlammedMyKindle** [comment]
bot, build io.github.iSlammedMyKindle.godotGem

**Comment by barthalion** [comment]
bot, build io.github.iSlammedMyKindle.godotGem

**Comment by barthalion** [comment]
```
stat: cannot statx 'builddir/*/share/app-info/xmls/io.github.iSlammedMyKindle.godotGem.xml.gz': No such file or directory
```

**Comment by barthalion** [comment]
To clarify the error above, it means that CI couldn't find appdata/metainfo file. See https://github.com/flathub/com.spotify.Client/blob/master/com.spotify.Client.appdata.xml for an example.

**Comment by hfiguiere** [comment]
it's also missing a .desktop file

File: io.github.iSlammedMyKindle.godotGem.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - --allow=bluetooth
+  - --device=all
+modules:
+  - name: desktopEntry
+    buildsystem: simple
+    build-commands:
+      - mkdir $FLATPAK_DEST/share
+      - mv * $FLATPAK_DEST/share
+    sources:
+      - type: dir
```
Comment (hfiguiere) [inline io.github.iSlammedMyKindle.godotGem.yml:20]: please don't use `dir`. list the individual files and use `install` to install them. Also this should be part of the main module.

File: io.github.iSlammedMyKindle.godotGem.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+app-id: io.github.iSlammedMyKindle.godotGem
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: godotGemClient.x86_64
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline io.github.iSlammedMyKindle.godotGem.yml:8]: does it support wayland?

File: io.github.iSlammedMyKindle.godotGem.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  - --socket=pulseaudio
+  - --share=network
+  - --allow=bluetooth
```
Comment (hfiguiere) [inline io.github.iSlammedMyKindle.godotGem.yml:11]: why do you need bluetooth?

File: io.github.iSlammedMyKindle.godotGem.yml (hunk 2 above)
Comment (iSlammedMyKindle) [reply io.github.iSlammedMyKindle.godotGem.yml:8]: Godot (the engine that this project is using) doesn't support wayland yet. It looks like regardless if I use this flag or not, it's running on X11. I'll remove this flag for you

File: io.github.iSlammedMyKindle.godotGem.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    build-commands:
+      - mkdir $FLATPAK_DEST/share
+      - mv * $FLATPAK_DEST/share
+    sources:
+      - type: dir
+        path: share
+  - name: godotGem
+    buildsystem: simple
+    build-commands:
+      - install -D godotGemClient.x86_64 $FLATPAK_DEST/bin/godotGemClient.x86_64
```
[the remark this answers is not in the document]
Comment (iSlammedMyKindle) [reply io.github.iSlammedMyKindle.godotGem.yml:25]: Added a `flatpak.json` in the next commit :+1:

**Comment by iSlammedMyKindle** [comment]
bot, build io.github.iSlammedMyKindle.godotGem

**Comment by iSlammedMyKindle** [comment]
bot, build io.github.iSlammedMyKindle.godotGem

**Comment by iSlammedMyKindle** [comment]
bot, build io.github.iSlammedMyKindle.godotGem

**Comment by iSlammedMyKindle** [comment]
Tested the app and it's working!

File: io.github.iSlammedMyKindle.godotGem.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        url: https://github.com/iSlammedMyKindle/godotGem/releases/download/1.0/godotGemClient.zip
+        sha256: 8a399df2e696dca134038c3bb79a890f74d65f65a2872bca072c30562ccd21ad
+      - type: file
+        path: icon.png
+      - type: file
+        path: icon.svg
+      - type: file
+        path: io.github.iSlammedMyKindle.godotGem.desktop
+      - type: file
+        path: io.github.iSlammedMyKindle.godotGem.metainfo.xml
```
Comment (barthalion) [inline io.github.iSlammedMyKindle.godotGem.yml:31]: Please consider making these files part of the zip file above in the future releases.

**Comment by iSlammedMyKindle** [comment]
Thank you so much!

**Pull request closed without merging**